### ISSUE TITLE
Memory Usage Too High

### ISSUE THREAD

**Issue by lawrenceleejr** [rsmith54/RJigsawTools#123]
Hey @rsmith54,

If you get a chance, can you try to understand why this uses so much memory? Jobs are failing on the grid because of it:

http://bigpanda.cern.ch/task/9541417/

e.g. 

```
skip ANALY_ZA-WITS-CORE due to site RAM shortage=2000(site upper limit) < 5700
```

So it wants to be using ~6 GB of RAM for some reason. I'll also try to investigate, but I don't see where it could be going wrong right now...

I see that we have a memory leak somewhere (memory usage creeps up) but I don't understand how the memory is handled in this framework well enough to know what's happening. (Not as simple as explicit missing `delete` calls in our code...)

-L

**Comment by lawrenceleejr** [comment]
Or actually maybe it is... I'm going to try and clean some stuff up.

-L

**Comment by lawrenceleejr** [comment]
I've narrowed it down to the RJigsaw calculator. Removing that algorithm from the job leads to stable memory usage.

**Comment by lawrenceleejr** [comment]
I no longer see a memory leak and feel like a crazy person. I guess the scouting job is what measures the required memory? I'm back at square one.

**Comment by lawrenceleejr** [comment]
Ok I can see that if I comment out all of the algorithms except BasicEventSelection, there's no memory growth. The memory leak starts with CalibrateST.

In the execute call of CalibrateST, I've narrowed it down to the Get\* calls to ST:

https://github.com/rsmith54/FactoryTools/blob/master/Root/CalibrateST.cxx#L207

Once L207-246 are uncommented, the leaking starts...

Maybe this is something we need to bring up with Martin+Zach?

Investigating.

-L

**Comment by lawrenceleejr** [comment]
Haha! Appears to be taus. Damn taus...

I'm going to comment out the GetTau call in CalibrateST and test on the grid.

**Comment by lawrenceleejr** [comment]
this appears to be entirely from the GetTaus() call in the CalibrateST alg.

Commenting out this line appears to remove the memory leak. 

```
Memory usage on lxplus node after 10k events
Only BasicEventSelection+CalibrateST (w GetTaus() call) - 4.7%
Only BES+ST (no taus) - 3.1%

Entire job (no taus) - 3.1% 
```

Without the leak, should be able to run stably within 1 GB of RAM.

**Comment by lawrenceleejr** [comment]
Fixed in #124
